stream-chat-android-ui-components/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt

```diff
@@ -405,6 +407,50 @@ public class MessageInputViewModel @JvmOverloads constructor(
         }
     }
 
+    /**
+     * Queries the backend for channel members whose username contains the string represented by the argument
+     * [contains].
+     *
+     * @param contains The string for which we are querying the backend in order to see if it is contained
+     * within a member's username.
+     *
+     * @return A list of users whose username contains the string represented by [contains] or an empty list in case
+     * no usernames contain the given string.
+     */
+    internal suspend fun queryMembersByUserNameContains(
+        contains: String,
+    ): List<User> {
+        logger.v { "[queryMembersByUserNameContains] Querying the backend for members." }
+
+        val (channelType, channelId) = cid.cidToTypeAndId()
+
+        val result = chatClient.queryMembers(
+            channelType = channelType,
+            channelId = channelId,
+            offset = QUERY_MEMBERS_REQUEST_OFFSET,
+            limit = QUERY_MEMBERS_REQUEST_MEMBER_LIMIT,
+            filter = Filters.autocomplete(
+                fieldName = "name",
+                value = contains
+            ),
+            sort = QuerySortByField(),
+            members = listOf()
+        ).await()
+
+        return if (result.isSuccess) {
+            result.data().map { it.user }.filter { it.name.contains(contains, true) }
```

Comment:
I prefer:
```
result.data().filter { it.user.name.contains(contains, true) }.map { it.user }
```
or:
```
result.data().asSequence().map { it.user }.filter { it.name.contains(contains, true) }.toList()
```

Because there's no need to iterate in all the elements of the list if you're going to filter it after. You gain a bit of performance with the options above with very little code change.

Reply:
Great point!

Reply:
@filbabic indentation suggestion is also valid.

Reply:
Applied first suggestion along with @filbabic's indentation suggestion. :smile: